pkg/pool/pool.go

```diff
@@ -42,8 +42,9 @@ func NewBytesPool(minSize, maxSize int, factor float64, maxTotal uint64) (*Bytes
 		buckets:  make([]sync.Pool, len(sizes)),
 		sizes:    sizes,
 		maxTotal: maxTotal,
-		new: func(sz int) []byte {
-			return make([]byte, 0, sz)
+		new: func(sz int) *[]byte {
```

Comment:
Are we sure about pointer to bite slice?

slice is a pointer interally, right?

Reply:
Well it's a pointer to array but not to slice header (len, cap). But yea, interested to know more why that change. Looks wrong especialy that we allocate new slice so no point in having pointer.

Reply:
@bwplotka @povilasv Yes, this is to expected because we want to avoid storing the length and capacity of the slice so we should use pointers to slice as per this recommendation here http://staticcheck.io/docs/checks#SA6002. Do you disagree with it?

Reply:
Anyway, made two simple tests so that we could put this matter to rest:
https://play.golang.org/p/nWaPErUmKM9
https://play.golang.org/p/7tSpZYMhjNa

You can see that with ptrs the bytes of allocated heap objects is a bit lower.